pkg/services/action.go

```diff
@@ -42,15 +42,48 @@ func NewTakeAction(
 	}
 }
 
+// NewSaveAction adds fields to an action and saves it in the db
+func NewSaveAction(
+	createAction func(context.Context, *models.Action) (*models.Action, error),
+	fetchUserInfo func(context.Context, string) (*models.UserInfo, error),
+) func(context.Context, *models.Action) error {
+	return func(ctx context.Context, action *models.Action) error {
+		actorInfo, err := fetchUserInfo(ctx, appcontext.Principal(ctx).ID())
+		if err != nil {
+			return err
+		}
+		if actorInfo == nil || actorInfo.Email == "" || actorInfo.CommonName == "" || actorInfo.EuaUserID == "" {
```

Comment:
Non-blocking suggestion: we seem to do this a bunch... Could this empty-value check be moved into the `cedarldap` client?